From 5c8d1e40259c3284b5deb4ae2943ccc95ad277a6 Mon Sep 17 00:00:00 2001
From: Sergey Fedorov <barracuda@macos-powerpc.org>
Date: Sat, 6 Sep 2025 16:28:12 +0800
Subject: [PATCH] scard.i: define SCARD_PROTOCOL_UNDEFINED when undefined

SCARD_PROTOCOL_UNDEFINED is not defined in Mac OS X Snow Leopard.
---
 src/smartcard/scard/scard.i | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/smartcard/scard/scard.i b/src/smartcard/scard/scard.i
index c3577062..a31dc282 100644
--- a/src/smartcard/scard/scard.i
+++ b/src/smartcard/scard/scard.i
@@ -132,6 +132,12 @@ Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
 
 #ifdef __APPLE__
 #include <PCSC/winscard.h>
+
+// Mac OS X Snow Leopard
+#ifndef SCARD_PROTOCOL_UNDEFINED
+#define SCARD_PROTOCOL_UNDEFINED 0x00000000
+#endif
+
 #else
 #include <winscard.h>
 #endif
